src/caretogether-pwa/src/Families/AssignmentCard.tsx

```diff
@@ -0,0 +1,179 @@
+import { Card, CardContent, Typography, Box } from '@mui/material';
+import PersonPinCircleIcon from '@mui/icons-material/PersonPinCircle';
+import CalendarTodayIcon from '@mui/icons-material/CalendarToday';
+import { ArrangementEntry, PartneringFamilyInfo } from '../GeneratedClient';
+import {
+  usePersonAndFamilyLookup,
+  useFamilyLookup,
+} from '../Model/DirectoryModel';
+import { useLoadable } from '../Hooks/useLoadable';
+import { partneringFamiliesData } from '../Model/ReferralsModel';
+import { useAppNavigate } from '../Hooks/useAppNavigate';
+import { format } from 'date-fns';
+import { FamilyName } from './FamilyName';
+
+interface AssignmentCardProps {
+  assignment: ArrangementEntry;
+}
+
+function allArrangements(partneringFamilyInfo: PartneringFamilyInfo) {
+  const closedReferralArrangements =
+    partneringFamilyInfo.closedReferrals?.flatMap(
+      (referral) =>
+        referral.arrangements?.map((arrangement) => ({
+          referralId: referral.id!,
+          arrangement,
+        })) ?? []
+    ) ?? [];
+
+  const openReferralArrangements =
+    partneringFamilyInfo.openReferral?.arrangements?.map((arrangement) => ({
+      referralId: partneringFamilyInfo.openReferral!.id!,
+      arrangement,
+    })) ?? [];
+
+  return [...closedReferralArrangements, ...openReferralArrangements];
+}
+
+export function AssignmentCard({ assignment }: AssignmentCardProps) {
+  const personAndFamilyLookup = usePersonAndFamilyLookup();
+  const familyLookup = useFamilyLookup();
+  const partneringFamilies = useLoadable(partneringFamiliesData);
+  const navigate = useAppNavigate();
+
+  const isCompleted = assignment.endedAtUtc !== undefined;
+  const progressWidth = isCompleted ? '100%' : '50%';
+  const statusColor = isCompleted ? '#2E7D32' : '#E3AE01';
+
+  const personInfo = assignment.partneringFamilyPersonId
+    ? personAndFamilyLookup(assignment.partneringFamilyPersonId)
+    : null;
+
+  const latestLocationId = assignment.childLocationHistory?.length
+    ? assignment.childLocationHistory[
+        assignment.childLocationHistory.length - 1
+      ].childLocationFamilyId
+    : null;
+
+  const latestLocationFamily = latestLocationId
+    ? familyLookup(latestLocationId)
+    : null;
+
+  const nextPlannedLocationId = assignment.childLocationPlan?.length
```

Comment:
Remove `nextPlannedLocationId` since the `nextPlannedLocation` (below) already does the same search. Anywhere that you were using `nextPlannedLocationId`, just use `nextPlannedLocation?.childLocationFamilyId`